lib/gst/player/gstplayer.c

```diff
@@ -1787,6 +1818,105 @@ subtitle_changed_cb (GObject * object, gpointer user_data)
   g_mutex_unlock (&self->lock);
 }
 
+static void *
+get_title (GstTagList *tags)
+{
+  gchar *title = NULL;
+
+  gst_tag_list_get_string (tags, GST_TAG_TITLE, &title);
+  if (!title)
+    gst_tag_list_get_string (tags, GST_TAG_TITLE_SORTNAME, &title);
+
+  return title;
+}
+
+static void *
+get_container (GstTagList *tags)
+{
+  gchar *container = NULL;
+
+  gst_tag_list_get_string (tags, GST_TAG_CONTAINER_FORMAT, &container);
```

Comment:
As a fallback, you could also get the caps from the source's srcpad (playbin's "source" property). It might contain caps. Or check the file extension in the URI if any. Maybe just add a TODO comment here for the future